projects/gloo/pkg/translator/listener.go

```diff
@@ -222,11 +222,18 @@ func validateListenerPorts(proxy *v1.Proxy, listenerReport *validationapi.Listen
 
 func newSslFilterChain(downstreamConfig *envoyauth.DownstreamTlsContext, sniDomains []string, useProxyProto *types.BoolValue, listenerFilters []*envoylistener.Filter) *envoylistener.FilterChain {
 
+	// copy listenerFilter so we can modify filter chain later without changing the filters on all of them!
+	listenerFiltersCopy := make([]*envoylistener.Filter, len(listenerFilters))
+	for i, lf := range listenerFilters {
+		tmp := *lf
+		listenerFiltersCopy[i] = &tmp
```

Comment:
Any reason why you did not do
```suggestion
		tmp := proto.Clone(lf).(*envoylistener.Filter)
		listenerFiltersCopy[i] = tmp
```